addons/project/models/project_project.py

```diff
@@ -208,11 +208,26 @@ def _read_group_stage_ids(self, stages, domain, order):
     milestone_count = fields.Integer(compute='_compute_milestone_count', groups='project.group_project_milestone')
     milestone_count_reached = fields.Integer(compute='_compute_milestone_reached_count', groups='project.group_project_milestone')
     is_milestone_exceeded = fields.Boolean(compute="_compute_is_milestone_exceeded", search='_search_is_milestone_exceeded')
+    milestone_progress = fields.Integer("Milestones Reached", compute='_compute_milestone_reached_count')
+    next_milestone_id = fields.Many2one('project.milestone', compute='_compute_next_milestone_id', store=True)
+    is_next_milesotone_unreached_and_late = fields.Boolean(compute='_compute_is_next_milesotone_unreached_and_late')
 
     _sql_constraints = [
         ('project_date_greater', 'check(date >= date_start)', "The project's start date must be before its end date.")
     ]
 
+    @api.depends('milestone_ids', 'milestone_ids.is_reached', 'milestone_ids.deadline')
+    def _compute_next_milestone_id(self):
+        for project in self:
+            next_milestones = project.milestone_ids.filtered(lambda m: not m.is_reached)
+            project.next_milestone_id = next_milestones and next_milestones[0].id
```

Comment:
```suggestion
            project.next_milestone_id = project.milestone_ids.filtered(
                lambda m: not m.is_reached
            )[:1].id
```
But is sounds a little too costly to filter (1) in a loop, (2) if we only take the first result.
Do you think of another solution?

Reply:
What about this? Isn't a _read_group better, performance-wise?

Reply:
Changes Done!